frontend/subrequests/lint/lint.go

```diff
@@ -125,46 +118,38 @@ func (results *LintResults) ToResult() (*client.Result, error) {
 }
 
 func PrintLintViolations(dt []byte, w io.Writer) error {
-	var warnings LintResults
+	var results LintResults
 
-	if err := json.Unmarshal(dt, &warnings); err != nil {
+	if err := json.Unmarshal(dt, &results); err != nil {
 		return err
 	}
 
-	// Here, we're grouping the warnings by rule name
-	lintWarnings := make(map[string][]Warning)
-	lintWarningRules := []string{}
-	for _, warning := range warnings.Warnings {
-		if _, ok := lintWarnings[warning.RuleName]; !ok {
-			lintWarningRules = append(lintWarningRules, warning.RuleName)
-			lintWarnings[warning.RuleName] = []Warning{}
+	sort.Slice(results.Warnings, func(i, j int) bool {
+		sourceInfoI := results.Sources[results.Warnings[i].Location.SourceIndex]
```

Comment:
Same in here. As this data is coming from the remote side, lets validate that the various nil values or invalid sourceIndex does not cause panics.